src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.PropertySetDataFlowOperationVisitor.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis
+{
+    using PropertySetAnalysisData = IDictionary<AbstractLocation, PropertySetAbstractValue>;
+
+    internal partial class PropertySetAnalysis
+    {
+        /// <summary>
+        /// Operation visitor to flow the location validation values across a given statement in a basic block.
+        /// </summary>
+        private sealed class PropertySetDataFlowOperationVisitor :
+            AbstractLocationDataFlowOperationVisitor<PropertySetAnalysisData, PropertySetAnalysisContext, PropertySetAnalysisResult, PropertySetAbstractValue>
+        {
+            private const int MaxInterproceduralCallChain = 1;
+            private readonly ImmutableDictionary<IInvocationOperation, PropertySetAbstractValue>.Builder _hazardousUsageBuilderOpt;
+            private INamedTypeSymbol DeserializerTypeSymbol;
```

Comment:
>private INamedTypeSymbol DeserializerTypeSymbol [](start = 12, length = 47)

Can this be part of the analysis context? You probably don't want to recompute this every time for each interprocedural call...

Reply:
It could be.  Right now I'm getting its value from a dictionary lookup.

Reply:
Given that the actual name is part of the context, I think your current implementation is also fine.

Reply:
Leaving as is.